src/sidebar/components/menu-item.js

```diff
@@ -0,0 +1,157 @@
+'use strict';
+
+const classnames = require('classnames');
+const { createElement } = require('preact');
+const propTypes = require('prop-types');
+
+const { onActivate } = require('../util/on-activate');
+
+const SvgIcon = require('./svg-icon');
+
+/**
+ * An item in a dropdown menu.
+ *
+ * Dropdown menu items display an icon, a label and can optionally have a submenu
+ * associated with them.
+ *
+ * When clicked, menu items either open an external link, if the `href` prop
+ * is provided, or perform a custom action via the `onClick` callback.
+ *
+ * The icon can either be an external SVG image, referenced by URL, or a named
+ * icon rendered by an `SvgIcon`.
+ *
+ * For items that have submenus, the `MenuItem` only provides an indicator as
+ * to whether the submenu is open. The container is responsible for displaying
+ * the submenu items beneath the current item.
+ */
+function MenuItem({
+  href,
+  icon,
+  iconAlt,
+  isDisabled,
+  isExpanded,
+  isSelected,
+  isSubmenuVisible,
+  label,
+  onClick,
+  onToggleSubmenu,
+  style,
+}) {
+  const iconClass = 'menu-item__icon';
+  const iconIsUrl = icon && icon.indexOf('/') !== -1;
```

Comment:
Oh, it is possible. A React Element (the result of `<SomeComponent .../>`) is just a value so you can use it wherever you might use a JS value. So it would be an option to allow a `MenuItem` to accept an element as an icon (`<MenuItem icon={<SvgIcon .../>}`). So far I haven't found a need for this flexibility, but it may turn out to make sense in future.